src/moonlink/src/storage/wal.rs

```diff
@@ -1 +1,332 @@
 pub(crate) mod wal_persistence_metadata;
+use crate::row::MoonlinkRow;
+use crate::storage::filesystem::accessor::base_filesystem_accessor::BaseFileSystemAccess;
+use crate::storage::filesystem::accessor::filesystem_accessor::FileSystemAccessor;
+use crate::storage::filesystem::filesystem_config::FileSystemConfig;
+use crate::table_notify::TableEvent;
+use crate::Result;
+use futures::stream::{self, Stream};
+use futures::{future, StreamExt};
+use serde::{Deserialize, Serialize};
+use std::pin::Pin;
+use std::sync::Arc;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub(crate) enum WalEventEnum {
+    Append {
+        row: MoonlinkRow,
+        xact_id: Option<u32>,
+        is_copied: bool,
+    },
+    Delete {
+        row: MoonlinkRow,
+        xact_id: Option<u32>,
+    },
+    Commit {
+        xact_id: Option<u32>,
+    },
+    StreamAbort {
+        xact_id: u32,
+    },
+    StreamFlush {
+        xact_id: u32,
+    },
+}
+
+pub(crate) struct PersistAndTruncateResult {
+    pub file_persisted: Option<WalFileInfo>,
+    pub highest_deleted_file: Option<WalFileInfo>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct WalFileInfo {
+    pub file_number: u64,
+    pub highest_lsn: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub(crate) struct WalEvent {
+    pub lsn: u64,
```

Comment:
qq: why do we use `pub` for fields, but smaller visibility for struct `pub(crate)`?